### PR TITLE
executor: refactor union scan and dirty table (#11702) (#14057)

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
cherry-pick 
* #11702 
* #14057  ( cherry-pick this buf-fix, otherwise will fail in some unit-test).

### CODE REVIEW

**Comment by crazycs520** [comment]
@jackysp @coocood @djshow832 PTAL

**Comment by coocood** [comment]
LGTM

File: session/txn.go [hunk 1]
```diff
@@ -30,7 +30,6 @@ import (
 	"github.com/pingcap/tidb/store/tikv/oracle"
```
Comment (jackysp) [inline session/txn.go]: 企业微信截图_20200108101749

file permission changed

Comment (crazycs520) [reply session/txn.go]: great, done.

**Pull request merged**